lib/lib-storage/src/s3-transfer-manager/S3TransferManager.ts

```diff
@@ -84,29 +84,34 @@ export class S3TransferManager implements IS3TransferManager {
     callback: EventListener<TransferEvent>,
     options?: AddEventListenerOptions | boolean
   ): void;
-  public addEventListener(
-    type: string,
-    callback: EventListener | null,
-    options?: AddEventListenerOptions | boolean
-  ): void;
-  public addEventListener(
-    type: unknown,
-    callback: EventListener<Event>,
-    options?: AddEventListenerOptions | boolean
-  ): void {
+  public addEventListener(type: string, callback: EventListener, options?: AddEventListenerOptions | boolean): void;
+  public addEventListener(type: string, callback: EventListener, options?: AddEventListenerOptions | boolean): void {
     const eventType = type as keyof TransferEventListeners;
     const listeners = this.eventListeners[eventType];
 
     if (!listeners) {
       throw new Error(`Unknown event type: ${eventType}`);
     }
 
-    // TODO: Add support for AbortSignal
-
     const once = typeof options !== "boolean" && options?.once;
+    const signal = typeof options !== "boolean" ? options?.signal : undefined;
     let updatedCallback = callback;
+
+    if (signal?.aborted) {
+      return;
+    }
+    if (signal) {
+      signal.addEventListener(
+        "abort",
+        () => {
+          this.removeEventListener(eventType, updatedCallback);
+        },
+        { once: true }
+      );
+    }
```

Comment:
- [ ] on abort the method-local event listeners need to be removed
- [ ] when the request completes on-abort listener itself needs to be removed